manifests/site.pp

```diff
@@ -15,8 +15,10 @@
 # More info: https://docs.puppet.com/puppet/latest/reference/ssl_attributes_extensions.html
 # You may need to change and adapt them according to your hiera.yaml
 # You can keep them also if you don't set extended trusted facts.
-if $ec2_tag_role {
-  $role = $ec2_tag_role
+if !defined('$role') {
```

Comment:
This is ok, but note that if someone sets an external fact called $role then that one is used instead of ec2_tag_role. Which maybe it's exactly what you want, just be aware of that